synchronization/src/cyface/java/de/cyface/synchronization/CyfaceHttpConnection.java

```diff
@@ -172,10 +154,58 @@ private HttpResponse readResponse(final @NonNull HttpURLConnection con)
                     throw new DataTransmissionException(response.getResponseCode(), "unknown response attributes",
                             response.getBody().toString());
                 }
+            } catch (final JSONException e) {
+                throw new ResponseParsingException(
+                        String.format("'%s'. Unable to read the http response.", e.getMessage()), e);
             }
-        } catch (final IOException | JSONException e) {
-            throw new ResponseParsingException(
-                    String.format("Error: '%s'. Unable to read the http response.", e.getMessage()), e);
         }
     }
+
+    /**
+     * Extracts the {@link HttpResponse} from the {@link HttpURLConnection}.
+     *
+     * @param con the {@link HttpURLConnection} to read the response from
+     * @return the {@link HttpResponse}
+     * @throws ResponseParsingException when the server response was unreadable
+     * @throws SynchronisationException when a connection error occurred while reading the response code
+     */
+    private HttpResponse readResponseFromConnection(final HttpURLConnection con)
+            throws ResponseParsingException, SynchronisationException {
+        String responseString;
+        try {
+            responseString = readInputStream(con.getInputStream());
+        } catch (final IOException e) {
+            // This means that an error occurred, read the error from the ErrorStream
+            // see https://developer.android.com/reference/java/net/HttpURLConnection
+            try {
+                responseString = readInputStream(con.getErrorStream());
+            } catch (IOException e1) {
+                throw new IllegalStateException("Unable to read error body.", e);
+            }
+        }
+
+        try {
+            return new HttpResponse(con.getResponseCode(), responseString);
+        } catch (final IOException e) {
+            throw new SynchronisationException("A connection error occurred while reading the response code.", e);
+        }
+    }
+
+    /**
+     * Reads a String from an InputStream.
+     *
+     * @param inputStream the {@link InputStream} to read from
+     * @throws IOException if an IO error occurred
+     * @return the {@link String} read from the InputStream
+     */
+    private String readInputStream(final InputStream inputStream) throws IOException {
+        final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+        final StringBuilder responseString = new StringBuilder();
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            responseString.append(line);
+        }
+        bufferedReader.close();
```

Comment:
Das hier ist ein mögliches Speicherleck, weil der bufferedReader nicht in einem finally geschlossen wird.